windows/internal/build_magma.bat

```diff
@@ -0,0 +1,101 @@
+
+:: Download Ninja
+curl -k https://s3.amazonaws.com/ossci-windows/ninja_1.8.2.exe --output C:\Tools\ninja.exe
+
+set "PATH=C:\Tools;C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v%CUVER%\bin;C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v%CUVER%\libnvvp;%PATH%"
+set CUDA_PATH=C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v%CUVER%
+set NVTOOLSEXT_PATH=C:\Program Files\NVIDIA Corporation\NvToolsExt
+
+mkdir magma_cuda%CUVER_NODOT%
+cd magma_cuda%CUVER_NODOT%
+
+:: First install MKL, which provides BLAS and LAPACK API
+:: Download and install from:
+:: curl https://s3.amazonaws.com/ossci-windows/w_mkl_2018.2.185.exe -k -O
+:: .\w_mkl_2018.2.185.exe
+:: Follow the installer steps and install MKL to default path
+
+pushd %CD%
+call "C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Auxiliary\Build\vcvarsall.bat" x86_amd64
+popd
+
+if not exist magma (
+  :: MAGMA 2.5.4 from http://icl.utk.edu/projectsfiles/magma/downloads/ with applied patches from our magma folder
+  git clone https://github.com/peterjc123/magma.git magma
+) else (
+  rmdir /S /Q magma\build
+  rmdir /S /Q magma\install
+)
+
+IF "%CUVER_NODOT%" == "80" (
+  set "CUDAHOSTCXX=C:/Program Files (x86)/Microsoft Visual Studio 14.0/VC/bin/x86_amd64/cl.exe"
+) ELSE (
+  set CUDAHOSTCXX=
+)
+
+set MKLROOT=
+
+cd magma
+mkdir build && cd build
+
+IF "%CUVER_NODOT%" == "80" (
+  set GPU_TARGET=sm_35 sm_50 sm_52 sm_37 sm_53 sm_60 sm_61
+) ELSE (
+  set GPU_TARGET=All
+)
+
+set CUDA_ARCH_LIST= -gencode arch=compute_37,code=sm_37 -gencode arch=compute_50,code=sm_50 -gencode arch=compute_60,code=sm_60 -gencode arch=compute_70,code=sm_70
+
+IF "%CUVER_NODOT%" == "110" (
+  set "CUDA_ARCH_LIST=%CUDA_ARCH_LIST% -gencode arch=compute_80,code=sm_80"
+)
+
+IF "%CUVER_NODOT%" == "111" (
+  set "CUDA_ARCH_LIST=%CUDA_ARCH_LIST% -gencode arch=compute_80,code=sm_80 -gencode arch=compute_86,code=sm_86"
+)
+
+IF "%CUVER_NODOT%" == "112" (
+  set "CUDA_ARCH_LIST=%CUDA_ARCH_LIST% -gencode arch=compute_80,code=sm_80 -gencode arch=compute_86,code=sm_86"
+)
+
+IF "%CUVER_NODOT%" == "113" (
+  set "CUDA_ARCH_LIST=%CUDA_ARCH_LIST% -gencode arch=compute_80,code=sm_80 -gencode arch=compute_86,code=sm_86"
+)
+
+set CC=cl.exe
+set CXX=cl.exe
+
+cmake .. -DGPU_TARGET="%GPU_TARGET%" ^
+            -DUSE_FORTRAN=0 ^
+            -DCMAKE_CXX_FLAGS="/FS /Zf" ^
+            -DCMAKE_BUILD_TYPE=%CONFIG% ^
+            -DCMAKE_GENERATOR=Ninja ^
+            -DCMAKE_INSTALL_PREFIX=..\install\ ^
+            -DCUDA_ARCH_LIST="%CUDA_ARCH_LIST%"
+if errorlevel 1 exit /b 1
+
+cmake --build . --target install --config %CONFIG% -- -j%NUMBER_OF_PROCESSORS%
+if errorlevel 1 exit /b 1
+
+cd ..\..\..
+
+:: Create
+7z a magma_%MAGMA_VERSION%_cuda%CUVER_NODOT%_%CONFIG_LOWERCASE%.7z %cd%\magma_cuda%CUVER_NODOT%\magma\install\*
+
+:: Push to AWS
+aws s3 cp magma_%MAGMA_VERSION%_cuda%CUVER_NODOT%_%CONFIG_LOWERCASE%.7z %OSSCI_WINDOWS_S3% --acl public-read
```

Comment:
Perhaps it'd be good to gate this behind an env variable so that we can run this script without it actually pushing the archive into S3?

Similar to the `WITH_PUSH` variable done in the `manywheel/build_docker.sh` scripts?

Reply:
That's a really good point